artemis-commons/src/main/java/org/apache/activemq/artemis/utils/SecureHashProcessor.java

```diff
@@ -33,6 +33,6 @@ public String hash(String plainText) throws Exception {
    //storedValue must take form of ENC(...)
    public boolean compare(char[] inputValue, String storedValue) {
       String storedHash = storedValue.substring(4, storedValue.length() - 1);
-      return codec.verify(inputValue, storedHash);
+      return codec.compare(inputValue, storedHash);
```

Comment:
Ahhh. I've missed that the verify method isnt new, its an existing method defined in DefaultSensitiveStringCodec that you just implemented in another bit of DefaultSensitiveStringCodec. But added the essentially same compare method to the SensitiveDataCodec interface.

My main thought is still why have both, and why does the private CodecAlgorithm class need to exist within DefaultSensitiveStringCodec at all, just to define methods that are basically already defined on the SensitiveDataCodec interface.

Reply:
I have changed SecureHashProcessor to relay on  SensitiveDataCodec<String> to compare user passwords but SensitiveDataCodec<String> hadn't a compare/verify method. So I have added the compare method in SensitiveDataCodec and I have implemented it in DefaultSensitiveStringCodec calling the existing verify method.

I see your point on DefaultSensitiveStringCodec we could change its implementation but that could be another PR.

Reply:
The existing stuff is already a little messy and this expands it and makes it more complicated to follow overall. It seems rather unnecessary when it could instead be making it nicer and simpler even while adding/exposing this functionality. The 'could be another PR' changes tends not to happen, just leaving the earlier stuff layering the onion.

Reply:
Fair enough, I'll move this PR back to draft and I'll revisit DefaultSensitiveStringCodec